angular/src/app/viewer/visualization/layout-picker/layout-picker.component.ts

```diff
@@ -1,33 +1,57 @@
-import { Component, Input, OnInit, Output, EventEmitter } from '@angular/core';
+import { Component, Input, Output, EventEmitter } from '@angular/core';
 
 @Component({
   selector: 'app-layout-picker',
   templateUrl: './layout-picker.component.html',
   styleUrls: ['./layout-picker.component.scss'],
 })
-export class LayoutPickerComponent implements OnInit {
-  @Input() selectedLayout!: any;
-
-  @Output() changeLayoutEmitter = new EventEmitter<string>();
+export class LayoutPickerComponent {
+  protected _layouts = [
+    { label: 'None', name: 'null', type: 'NullLayoutOptions' },
+    { label: 'Random', name: 'random', type: 'RandomLayoutOptions' },
+    { label: 'Preset', name: 'preset', type: 'PresetLayoutOptions' },
+    { label: 'Grid', name: 'grid', type: 'GridLayoutOptions' },
+    { label: 'Circle', name: 'circle', type: 'CircleLayoutOptions' },
+    {
+      label: 'Concentric',
+      name: 'concentric',
+      type: 'ConcentricLayoutOptions',
+    },
+    {
+      label: 'Breadth-First',
+      name: 'breadthfirst',
+      type: 'BreadthFirstLayoutOptions',
+    },
+    { label: 'Cose', name: 'cose', type: 'CoseLayoutOptions' },
+  ] as const;
+  get layouts(): readonly Layout[] {
+    return this._layouts;
+  }
 
-  public layouts: any = [
-    { value: 'null', id: 'None' },
-    { value: 'custom', id: 'Custom' },
-    { value: 'random', id: 'Random' },
-    { value: 'grid', id: 'Grid' },
-    { value: 'circle', id: 'Circle' },
-    { value: 'concentric', id: 'Concentric' },
-    { value: 'breadthfirst', id: 'Breadth First' },
-    { value: 'cose', id: 'Force Directed' },
-  ];
+  private _disabled = true;
+  @Input()
+  set disabled(disabled: boolean) {
+    this._disabled = disabled;
+  }
+  get disabled(): boolean {
+    return this._disabled;
+  }
 
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
-  constructor() {}
+  private _currentLayout = 'Concentric';
+  get currentLayout(): string {
+    return this._currentLayout;
+  }
 
-  // eslint-disable-next-line @angular-eslint/no-empty-lifecycle-method, @typescript-eslint/no-empty-function
-  ngOnInit(): void {}
+  @Output() changeLayoutEmitter = new EventEmitter<string>();
 
-  changeLayout(value: any): void {
-    this.changeLayoutEmitter.emit(value);
+  changeLayout(options: Layout): void {
+    this._currentLayout = options.label;
+    this.changeLayoutEmitter.emit(options.name);
   }
 }
+
+interface Layout {
+  label: string;
+  name: string;
+  type: string;
```

Comment:
Is this `type` property going to be used in the future? Just wondering because I don't think it gets read anywhere.

Reply:
its used as the return type for the changeLayout function.. 
basically `any` cant be used as the type anything (I think) if you want to avoid lint errors..

Reply:
@RodolfoAllendes 
Isn't the return type of `changeLayout` void https://github.com/Toxygates/Panomicon/pull/19/files/95d601804075ad8a4b14a0a7736e4b6cdc9a59a1#diff-860375f887be92f6b7e531ec0394b882c2c3dd7e03e6fdd8a5fadda356f8aea2R47 ? Or maybe I am misunderstanding...

I did do a search for all references to the `type` property and didn't find any, but it could just be the editor + language system not being smart.

Also just to make sure, I'm not talking about the interface (`Layout`) is a whole, but the 'type: string' property, which I see is given a value whenever you declare instances of `Layout`, but which doesn't ever seem to be referred to.

Reply:
your right xD.. the return is void... the parameter the function gets `options: Layout` is the one that uses the interface..

as for the type property, that one gets emited as an event and is captured by the display-canvas component (parent component) that uses it to actually update the layout of the visualization.. it is a bit hard to follow as the link is done in the display-canvas.html file rather on the ts  (I had to look through to find it now too XD)

Reply:
@RodolfoAllendes 
Thanks for the clarification!

Sorry, can you do me a favor and link me to the location where the `type` property gets used? I just installed the Angular Language Service extension for VS Code so now "Find All References" does find references in HTML files as well, but I still wasn't able to find it :sweat_drops: 
(I also tried just searching for the string 'type' in the source and template files for LayoutPicker and DisplayCanvas but couldn't seem to find it...)

Reply:
@yujiberra of course I've been looking at this wrong the whole time.. thinking of the `name` rather than the `type` property, even when you made it clear twice already that it was the latter...

I remember including these at some point because the configurable options for each layout are different and, when using typescript, by specifying the type of options will give you access to them... then, I noticed that, with the level of detail that I am currently implementing, I probably dont need them, and hence, no reference (you were right since the beginning).

If not too bothered, I would like to keep them a bit longer hanging around though, just in case I do end up using them in the future..

Reply:
@RodolfoAllendes 
Thanks for the explanation, makes sense! 

> If not too bothered, I would like to keep them a bit longer hanging around though, just in case I do end up using them in the future..

Yup yup no problem at all!